application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,178 @@
+        MessageAction message = auditLogChannel.orElseThrow().sendMessageEmbeds(messageEmbed(user));
+
+        message = buildAttachment(attachments, message);
+        message.queue();
+
+        event.reply("Thank you for contacting the moderators").setEphemeral(true).queue();
+
+    }
+
+    private int getRemainingTime(MessageChannel messageChannel) {
+        return COOLDOWN_DURATION_VALUE - getTimePassedSinceCommandCalled(messageChannel);
+    }
+
+    private String getName(SlashCommandInteractionEvent event) {
+        String user = event.getUser().getAsMention();
+        boolean optionalMessage = event.getOption(OPTION_MESSAGE_PRIVATE).getAsBoolean();
+        if (optionalMessage) {
+            user = "Anonymous";
+        }
+        return user;
+    }
+
+    private MessageAction buildAttachment(List<ModAuditLogWriter.Attachment> attachments,
+            MessageAction message) {
+        for (ModAuditLogWriter.Attachment attachment : attachments) {
+            message = message.addFile(attachment.getContentRaw(), attachment.name());
+        }
+        return message;
+    }
+
+    private MessageEmbed messageEmbed(String user) {
+        return new EmbedBuilder().setAuthor("Modmail Command invoked")
+            .setColor(Color.BLACK)
+            .setTitle("/modmail from '%s' ".formatted(user))
+            .build();
+    }
+
+    private List<ModAuditLogWriter.Attachment> getAttachments(SlashCommandInteractionEvent event) {
+        String content = event.getOption(MESSAGE).getAsString();
+        List<ModAuditLogWriter.Attachment> attachments = new ArrayList<>();
+        attachments.add(new ModAuditLogWriter.Attachment("content.md", content));
+        return attachments;
+    }
+
+    private Optional<TextChannel> getChannel(SlashCommandInteractionEvent event) {
+        Guild guild = event.getJDA()
+            .getGuildCache()
+            .stream()
+            .filter(guildName -> guildNamePatternPredicate.test(guildName.getName()))
+            .toList()
+            .get(0);
+        if (guild == null) {
+            return Optional.empty();
+        }
+        return guild.getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    private boolean isChannelOnCooldown(MessageChannel channel) {
+        return Optional
+            .ofNullable(channelIdToLastCommandInvocation.getIfPresent(channel.getIdLong()))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    private int getTimePassedSinceCommandCalled(MessageChannel channel) {
+        return Optional
+            .ofNullable(channelIdToLastCommandInvocation.getIfPresent(channel.getIdLong()))
+            .map(this::getTime)
+            .orElse(COOLDOWN_DURATION_VALUE);
+    }
```

Comment:
Since COOLDOWN_DURATION_VALUE is int and the getTime was returning long

Reply:
i understand. but u can most likely do all of that while staying within `Duration`. The only reason the `COOLDOWN_DURATION_VALUE` is `int` and not `Duration` is because the scheduler doesnt take `Duration` but takes those two split.

Anyways, with my suggestion to remove the "remaining time" feature, this whole stuff goes away regardless :)

Reply:
Done